### PR TITLE
check for mips targets for stat.st_dev definitions

### PR DESCRIPTION
**Pull request by kraj**
See
https://sourceware.org/bugzilla/show_bug.cgi?id=17786

Signed-off-by: Khem Raj <raj.khem@gmail.com>

### CODE REVIEW

File: src/logging.rs [hunk 1]
```diff
@@ -292,7 +292,10 @@ impl JournalStream {
     pub fn from_fd<F: AsRawFd>(fd: F) -> std::io::Result<Self> {
         nix::sys::stat::fstat(fd.as_raw_fd())
             .map(|stat| JournalStream {
+                #[cfg(all(not(target_arch = "mips")))]
```
Comment (lucab) [inline src/logging.rs:295]: Here and below, I think the `all()` bit is spurious.

**Comment by lucab** [comment]
Thanks for the PR. Can you maybe add some details about where this showed up?

**Comment by lunaryorn** [comment]
I'd also appreciate a summary of the linked bug report; it's rather long, and I'm not sure what information or fact I should read from it.

**Comment by lunaryorn** [comment]
What Rust tier does the corresponding target belong to?

**Comment by lunaryorn** [comment]
I've opened #104 to use `std::fs` in this place instead of `libc`.

**Comment by lunaryorn** [comment]
@kraj Did you have a chance to test #104?

**Comment by kraj** [comment]
> @kraj Did you have a chance to test #104?

in limited build only test it seems to be ok.

**Comment by lucab** [comment]
@kraj thanks for the additional feedback. I'd also prefer taking the rework in #104 to address this.

**Comment by lunaryorn** [comment]
I'm closing this in favour of #104 which I just merged 🙂

**Pull request closed without merging**